### DISCUSSION TITLE
[ideas] Remove `pingr`dependency to help making `tar_watch` WASM compatible

### DISCUSSION THREAD

**Discussion by Adafede** [ropensci/targets#1317] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi @wlandau,

I recently tried building my targets shiny app (based on your https://github.com/wlandau/targets-shiny) available via `shinylive`. For it, it needs all dependencies to be [r-wasm](https://docs.r-wasm.org/webr/latest/) compatible.

It looks like `pingr` is not and thus the app is unable to run.

As it is only needed once for a small check, could we try removing it to see if we can port targets shiny apps to shinylive? 😊 

P.S.: 
If of any use here are the repos:
- https://github.com/taxonomicallyinformedannotation/tima
- https://github.com/taxonomicallyinformedannotation/taxonomicallyinformedannotation.r-universe.dev
- https://github.com/taxonomicallyinformedannotation/tima-shinylive

**Comment by hadley** [comment]
Alternatively, maybe we could add r-wasm shim for pingr. Let me see how hard that would be.

**Reply by wlandau** [reply]
Thanks for offering. For `targets`, I actually do like the approach in #1318. It hadn't occurred to me to try `socketConnection()`, and it's fine for `tar_watch()` as an alternative to `pingr`.
